chanbackup/pubsub.go

```diff
@@ -95,8 +96,13 @@ type ChannelNotifier interface {
 // be used to implement a system that always keeps the multi-chan backup file
 // on disk in a consistent state for safety purposes.
 type SubSwapper struct {
-	started sync.Once
-	stopped sync.Once
+	// started tracks whether the SubSwapper has been started and ensures
+	// it can only be started once.
+	started atomic.Bool
+
+	// stopped tracks whether the SubSwapper has been stopped and ensures
+	// it can only be stopped once.
+	stopped atomic.Bool
```

Comment:
Can we remove `stopped` and just use `quit` state as an indication of closeness? These two variables seem to have the same meaning.